### DISCUSSION TITLE
[General] Quickstart - issue compiling contract in Quickstart documentation

### DISCUSSION THREAD

**Discussion by GavinStein1** [zkSync-Community-Hub/zksync-developers#49] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I am trying to follow the Quickstart documentation. I have completed all steps up to the contract compilation point, and I have met/satisfied all prerequisites of the Quickstart.

### What is the specific issue or error you're encountering?

When trying to run the `yarn hardhat compile` command, I get the following error:

`Error: zksync-web3 tried to access @ethersproject/hash, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.`

### Can you share the error messages or logs you're receiving, if any?

`Required package: @ethersproject/hash
Required by: zksync-web3@virtual:2ed73bbccb4ade5a7e201e8c85ec39f17bc355432387897490887cd33b2ce1946819b937fafb8462ed6f4247d1b59428bb859862a08d85394493cb8d3b8baafe#npm:0.14.4 (via /Users/<user>/zksync-greeter-example/.yarn/__virtual__/zksync-web3-virtual-605e97e0fd/0/cache/zksync-web3-npm-0.14.4-3181290eb1-f702a3437f.zip/node_modules/zksync-web3/build/src/)`

### Have you made any recent changes to the contract before encountering this issue?

I haven't made any changes to the code that is provided in the doc. 

### Are there any external libraries or contracts that your contract interacts with?

No external libraries besides the dependencies already known. 

### Can you provide the relevant portions of your contract code where the issue is occurring?

n/a

### Have you tried to isolate the problem, and if so, what were the results?

n/a

### What steps have you already taken to try to resolve the issue?

I would like to modify the zksync-web3 package to add @ethersproject/hash as a dependency but I am not sure where to do this.

**Comment by dutterbutter** [comment]
Hello 👋  

It seems like you're following the steps accurately and there should **not** be any additional dependencies required outside of those mentioned in the Quickstart step 2. I went through the quickstart myself from scratch and could not reproduce but lets try to dig deeper to see whats causing the issue for you. 

To better assist you, I have a few follow-up questions:

1. What environment are you running on? Knowing your operating system and its version could be helpful for further investigation.
2. When we get to the contract compilation step, there should be only the `/contracts/Greeter.sol` contract to be compiled. Can you confirm that this is the only contract present?
3. Could you share a structure of your project directory and `package.json` file?

**Reply by GavinStein1** [reply]
Hi Dustin, thanks for your response!

1. I am on MacOS 11.6.7. I used nvm to switch Node.js version to 16.16.0.
2. the Greeter contract is the only contract in the contracts dir.
3. My package.json file is below.
`{
  "name": "zksync-greeter-example",
  "packageManager": "yarn@3.6.1",
  "devDependencies": {
    "@matterlabs/hardhat-zksync-deploy": "^0.6.3",
    "@matterlabs/hardhat-zksync-solc": "^0.4.1",
    "@matterlabs/hardhat-zksync-verify": "^0.2.0",
    "@nomiclabs/hardhat-etherscan": "^3.1.7",
    "ethers": "^5.7.2",
    "hardhat": "^2.16.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6",
    "zksync-web3": "^0.14.3"
  }
}`

Something else that was inconsistent with the Quickstart is the following command:
`yarn add -D typescript ts-node ethers@^5.7.2 zksync-web3 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy @matterlabs/hardhat-zksync-verify @nomiclabs/hardhat-etherscan`
This errored and said ethers@^5.7.2 no match found. This was fixed by wrapping the package and version in quotes. I don't believe this has something to do with my current issue but more to do with running zsh in the terminal. 

I really appreciate your help with this, I'd love to get this working.

**Comment by GavinStein1** [comment]
I've just tried the quickstart steps again but using npm instead of yarn and it has worked (well I have successfully deployed the contract to the Goerli testnet). I am not able to isolate exactly what it is with yarn that was causing me issues. I am using yarn v3.6.1. 

Unrelated to this issue, but have zk-sync contracts been deployed to the Sepolia testnet to use that network instead. I'm not familiar with the intricacies of either but I find transaction speeds much faster and better for dev on the Sepolia network.

**Reply by dutterbutter** [reply, marked as answer]
That's great news! 😃 I'm glad to hear that the quickstart steps have worked with `npm`. I'll certainly look into this `yarn` issue further with the version (v3.6.1) you've specified. Your feedback is invaluable, thank you for bringing this to our attention! 🙏

Regarding the deployment to Sepolia, unfortunately, we haven't made that move yet. However, it's definitely on our radar! 📡 We're actively looking to support additional testnets in the near future. So, stay tuned! 🚀

**Comment by martinerinv** [comment]
It seems that there is a dependency resolution issue with zksync-web3 package. You can try resolving it by ensuring that all dependencies are correctly installed and compatible. Alternatively, you can reach out to the maintainers of zksync-web3 package for assistance in resolving this issue. They may provide guidance on how to address the dependency ambiguity or update their package to resolve the issue.

**Comment by evmeiwear** [comment]
The error indicates that zksync-web3 is trying to access @ethersproject/hash, but it's not declared in its dependencies. To resolve this, you may need to update zksync-web3 to a version that properly declares @ethersproject/hash as a dependency.

**Comment by mcleodjv** [comment]
It seems like there is a dependency issue with the zksync-web3 package trying to access @ethersproject/hash without it being declared in its dependencies. To resolve this, you can try updating the zksync-web3 package to a version that includes the required dependency, or you can manually add @ethersproject/hash as a dependency in the package.json file of your project. Make sure to install the necessary dependencies after making any changes to the package.json file.

**Comment by casesumteryan** [comment]
To resolve this issue, you can try adding the missing dependency directly to your project. Open your terminal and navigate to your project directory. Then, run the following command: yarn add @ethersproject/hash
This will add the required @ethersproject/hash package to your project's dependencies, which should resolve the ambiguity issue with zksync-web3. After adding the dependency, try running yarn hardhat compile again to see if the error persists.

**Comment by hyal1ngrumblu** [comment]
The user is encountering an error when running the yarn hardhat compile command while following the Quickstart documentation. The error message states that zksync-web3 tried to access @ethersproject/hash, but it isn't declared in its dependencies. The user has not made any recent changes to their contract code and is not interacting with any external libraries or contracts. The user is requesting assistance with modifying the zksync-web3 package to add @ethersproject/hash as a dependency.
